assets/pages/regular-contributions/regularContributions.jsx

```diff
@@ -47,9 +37,21 @@ const router = (
   <BrowserRouter>
     <Provider store={store}>
       <div>
-        <Route exact path={routes.recurringContribCheckout} component={RegularContributionsPage} />
-        <Route exact path={routes.recurringContribThankyou} component={ContributionsThankYouPageContainer} />
-        <Route exact path={routes.recurringContribPending} component={ContributionsThankYouPageContainer} />
+        <Route
+          exact
+          path={routes.recurringContribCheckout}
+          component={() => <RegularContributionsPage />}
```

Comment:
What's the story with this?

Reply:
Discussed offline, but here's the reasoning for anyone else who's curious.

A React stateless component, which is what we use exclusively in this codebase, is just a function. For example:

```jsx
function Hello(props: { name: string }) {
  return <p>Hello {props.name}</p>;
}

const pageContent = (
  <div>
    <Hello name="world" />
  </div>
);
```

Here the `</>` is just syntactic sugar for calling a function, so whenever these are used flow is able to treat that as the call-site for a function and check that the correct arguments (a.k.a. `props`) are being passed.

 React-Router's `Route` expects to be passed a component of type `React$ComponentType<*>` for its `component` prop, which it will then "call" somewhere along the way. For example:

```jsx
<Route path="/hello" component={Hello} />
```

Unfortunately, wherever this call-site is in the depths of React-Router, flow seems unable to figure out what the `props` should be, and is therefore unable to type-check them.

My solution is to create a wrapper component like so:

```jsx
const HelloWrapper = () => <Hello name="world" />;
```

This puts the call-site for the `Hello` component back into our code, which allows flow to type-check it. The wrapper component, which takes no props, is then called by React-Router.